java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1188,30 +965,21 @@ public TryStatementTreeImpl newTryWithResourcesStatement(
     }
   }
 
-  public ResourceListTreeImpl newResources(List<AstNode> rests) {
-    List<AstNode> children = Lists.newArrayList();
+  public ResourceListTreeImpl newResources(List<Tuple<VariableTreeImpl, Optional<InternalSyntaxToken>>> rests) {
     ImmutableList.Builder<VariableTreeImpl> resources = ImmutableList.builder();
 
-    for (AstNode rest : rests) {
-      VariableTreeImpl resource = (VariableTreeImpl) rest.getFirstChild();
-      children.add(resource);
+    for (Tuple<VariableTreeImpl, Optional<InternalSyntaxToken>> rest : rests) {
+      VariableTreeImpl resource = rest.first();
       resources.add(resource);
-
-      if (rest.getNumberOfChildren() == 2) {
-        children.add(rest.getLastChild());
-      }
+      //FIXME SONARJAVA-547 handle semi colon separator
     }
 
-    return new ResourceListTreeImpl(resources.build(), children);
+    return new ResourceListTreeImpl(resources.build());
   }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)